psi4/src/psi4/dfocc/prepare4grad.cc

```diff
@@ -954,5 +955,21 @@ void DFOCC::effective_pdm_gfm() {
     timer_off("Gamma^eff");
 }  // end effective_pdm_gfm
 
+void DFOCC::set_opdm() {
+    auto Da = std::make_shared<Matrix>("MO-basis alpha OPDM", nmo_, nmo_);
+    auto Db = std::make_shared<Matrix>("MO-basis beta OPDM", nmo_, nmo_);
+    if (reference_ == "RESTRICTED") {
+        G1->to_shared_matrix(Da);
+        Da->scale(0.5);
+        Da_ = linalg::triplet(Ca_, Da, Ca_, false, false, true);
+        Db_ = Da_;
+    } else {
+        G1A->to_shared_matrix(Da);
+        G1B->to_shared_matrix(Db);
+        Da_ = linalg::triplet(Ca_, Da, Ca_, false, false, true);
+        Db_ = linalg::triplet(Cb_, Db, Cb_, false, false, true);
```

Comment:
this is taking into account that ROHF is UNRESTRICTED in dfocc? https://github.com/psi4/psi4/blob/master/psi4/src/psi4/dfocc/dfocc.cc#L156-L157

Reply:
To work nicely with orbital optimization, `dfocc` is built to work with whatever orbitals your put in. The difference between UHF and ROHF is hidden in terms in the densities. That said, `dfocc` doesn't have ROHF densities anyways.

The MO densities are shamelessly stolen from formula [here](https://github.com/psi4/psi4/blob/eb2336bce2cb0c38fa6aeee35c8b3130cc5a5656/psi4/src/psi4/dfocc/properties.cc#L51), so those are right. The only question left is if the AO backtransformation is correct. And as long as Da and Db are the alpha and beta MO density matrices (see previous sentence), then those are correct too.